### PR TITLE
Lower IABS intrinsic.

### PR DESCRIPTION
**Pull request by Sameeranjoshi**

### CODE REVIEW

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 1]
```diff
@@ -1043,6 +1045,25 @@ mlir::Value IntrinsicLibrary::genAbs(mlir::Type resultType,
...
 }
 
+// IABS
+mlir::Value IntrinsicLibrary::genIAbs(mlir::Type resultType,
+                                      llvm::ArrayRef<mlir::Value> args) {
+  assert(args.size() == 1);
+  auto arg = args[0];
+  auto type = arg.getType();
+  if (auto intType = type.dyn_cast<mlir::IntegerType>()) {
+    // At the time of this implementation there is no abs op in mlir.
```
Comment (sscalpone) [inline flang/lib/Lower/IntrinsicCall.cpp:1055]: How about:

`// TODO Implement llvm integer abs intrinsic.`

**Comment by schweitzpgi** [comment]
I think the intention was to use the type agnostic ABS gen code. @jeanPerier

**Comment by jeanPerier** [comment]
> I think the intention was to use the type agnostic ABS gen code. @jeanPerier

That's correct, I would rather avoid duplicating the code for every specific version. As mentioned in the issue, I have a patch in mind that makes the mapping specific <-> generic name mapping accessible from the front-end and that will solve the issue for all unrestricted specific intrinsics with different name than the generic

**Comment by schweitzpgi** [comment]
Superseded by #290

**Pull request closed without merging**